SQF/dayz_code/compile/dze_buildChecks.sqf

```diff
@@ -73,5 +73,20 @@ if (_toolCheck) then {
 	_canBuild = [_item, _require, _classname] call dze_requiredItemsCheck;
 };
 
+_text = getText (configFile >> 'CfgMagazines' >> _item >> 'displayName');
+
+if (((count DZE_SafeZoneNoBuildItems) > 0) && {_classname in DZE_SafeZoneNoBuildItems}) then {
+	{
+		if ((player distance (_x select 0)) <  DZE_SafeZoneNoBuildDistance) exitWith { _canBuild = false; };
+	} forEach DZE_safeZonePosArray;
+	if !(_canBuild) exitWith { dayz_actionInProgress = false; format [localize "STR_EPOCH_PLAYER_166",_text,DZE_SafeZoneNoBuildDistance] call dayz_rollingMessages; [false, _isPole]; };
+};
+
+if (((count DZE_SafeZoneNoBuildNear) > 0) && {count (nearestObjects [_pos, DZE_SafeZoneNoBuildNear, DZE_BlacklistNoBuildDistance]) > 0}) exitWith {
+	dayz_actionInProgress = false;
+	format [localize "STR_EPOCH_PLAYER_167",_text,DZE_BlacklistNoBuildDistance] call dayz_rollingMessages;
```

Comment:
I would add a %3 to this saying what the blacklisted building type is, so the player knows if it is a barracks, control tower, etc. Just take the typeOf from the nearestObjects select 0 return.